client/pages/home.jsx

```diff
@@ -1,8 +1,81 @@
 import React from 'react';
+import AppContext from '../lib/app-context';
+import TripSelect from '../components/trip-select';
 
-export default function Home(props) {
-  return (
-    <>
-    </>
-  );
+export default class Home extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      trips: [],
+      trip: null,
+      tripSum: null,
+      tripBalance: null
+    };
+    this.handleSelect = this.handleSelect.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/trips')
+      .then(res => res.json())
+      .then(trips => {
+        this.setState({ trips });
+      });
+  }
+
+  handleSelect(event) {
+    fetch(`/api/trips/${event.target.value}`)
+      .then(res => res.json())
+      .then(data => {
+        const trip = data;
+        return fetch(`/api/expenses/${trip.tripId}`)
+          .then(res => res.json())
+          .then(expenses => {
+            if (expenses.length < 1) {
+              this.setState({ trip, tripSum: null, tripBalance: 0 });
+            } else {
+              const tripSum = expenses.reduce((accumulator, currentValue) => {
+                return accumulator + Number(currentValue.amount);
+              }, Number(expenses[0].amount)).toFixed(2);
+              const tripBalance = (Number(trip.budget) - tripSum).toFixed(2);
+              this.setState({ trip, tripSum, tripBalance });
+            }
+          });
+      })
```

Comment:
Would it be reasonable to re-work your `GET /api/trips/:tripId` route on the server to include its expenses for you? That way the client doesn't have to do any extra fetching and you can take advantage of some aggregate functions.